src/core/lib/security/credentials/ssl/ssl_credentials.cc

```diff
@@ -57,26 +79,62 @@ grpc_ssl_credentials::~grpc_ssl_credentials() {
     config_.verify_options.verify_peer_destruct(
         config_.verify_options.verify_peer_callback_userdata);
   }
+  tsi_ssl_client_handshaker_factory_unref(client_handshaker_factory_);
 }
 
 grpc_core::RefCountedPtr<grpc_channel_security_connector>
 grpc_ssl_credentials::create_security_connector(
     grpc_core::RefCountedPtr<grpc_call_credentials> call_creds,
     const char* target, grpc_core::ChannelArgs* args) {
+  if (config_.pem_root_certs == nullptr) {
+    gpr_log(GPR_ERROR, "Could not get default pem root certs.");
```

Comment:
nit: I think this error message is not correct, since it should not say "default". Could we change it to e.g. "No root certs."?

Reply:
Friendly ping :)

Reply:
Changed to `No root certs in config. Client-side security connector must have root certs.`